src/main/scala/rna16s.scala

```diff
@@ -91,29 +102,28 @@ case object rna16s extends AnyBlastDB {
 
   val uninformativeTaxIDs: Set[String] = uninformativeTaxIDsMap.keys.map(_.toString).toSet
 
-  /* Here we want to keep sequences which */
+  /*
+    #### Database defining predicate
+
+    Sequences that satisfy this predicate (on themselves together with their annotation) are included in this database.
+  */
+  import bio4jTaxonomyBundle._
   val predicate: (Row, FASTA.Value) => Boolean = { (row, fasta) =>
-    /* - are annotated as encoding 16S */
-    ( row.select(gene_name) == geneNameFor16S ||
-      fasta.getV(header).description.toLowerCase.contains("16s") ||
-      fasta.getV(header).description.toLowerCase.contains("small subunit ribosomal")
-    ) &&
     /* - are annotated as rRNA */
-     row.select(rna_type).toLowerCase.contains("rrna") &&
+     row.select(rna_type).contains(ribosomalRNAType)        &&
     /* - their taxonomy association is *not* one of those in `uninformativeTaxIDs` */
-    (uninformativeTaxIDs contains row.select(tax_id)) &&
-    /* - and the corresponding sequence is not shorter than 1300 BP */
-    (fasta.getV(sequence).value.length >= 1300) &&
+    ( ! uninformativeTaxIDs.contains(row.select(tax_id)) )  &&
+    /* - the corresponding sequence is not shorter than the threshold */
+    (fasta.getV(sequence).value.length >= minimum16SLength) &&
     /* - is a descendant of either Archaea or Bacteria */
-    bio4jTaxonomyBundle.checkAncestors(
-      row.select(tax_id), Set(
+    row.select(tax_id).isDescendantOfOneIn(
+      Set(
         "2157", // Archaea
         "2"     // Bacteria
```

Comment:
So length `1300` has to be a named constant, but these poor bacterias don't deserve it... 😝